### PR TITLE
Fix #2001: Disable submit button if any of the submission limit is exhausted

### LINKED ISSUE #2001: Disable Submit if Submission Limit is exhausted
**Deliverables**
- Disable Submit if any of the Submission Limit is exhausted
- Also, Submit Button should not be enables unless `Submission Upload File` is selected

This will decrease API Calls.

### PR DESCRIPTION
**Pull request by vkartik97**
Fixes: #2001 

Deliverables:
- [x] Disable Submit if any of the Submission Limit is exhausted
- [x] Submit Button should not be enables unless Submission Upload File is selected
<br>

#### Screenshots:

When Submission Limit is exhausted:
screenshot from 2019-01-16 02-16-36
<br>
When Submissions are remaining **but** `Upload File` is not selected:
screenshot from 2019-01-16 02-17-34
<br>
When Submissions are remaining **and** `Upload File` is selected:
screenshot from 2019-01-16 02-17-53

### CODE REVIEW

**Comment by vkartik97** [comment]
@deshraj @RishabhJain2018 Please Review

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -37,6 +37,7 @@
         vm.sortColumn = 'rank';
         vm.reverseSort = false;
         vm.columnIndexSort = 0;
+        vm.submissionsExhausted = true;
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeCtrl.js]: Please initialize with `false` value.

File: frontend/src/js/controllers/challengeCtrl.js [hunk 2]
```diff
@@ -1032,9 +1033,11 @@
                         else if (details.remaining_submissions_today_count > 0) {
                             vm.remainingSubmissions = details;
                             vm.showSubmissionNumbers = true;
+                            vm.submissionsExhausted = false;
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeCtrl.js]: Once you initialize it with `false`, you don't need this line. Please remove it.

**Comment by RishabhJain2018** [comment]
@vkartik97 Please let me know when you have incorporated the comments.

File: frontend/src/js/controllers/challengeCtrl.js (hunk 1 above)
Comment (vkartik97) [reply frontend/src/js/controllers/challengeCtrl.js]: @RishabhJain2018 
This is for when submissions are remaining **but** `Upload File` is not selected
screenshot from 2019-01-16 02-17-34

File: frontend/src/js/controllers/challengeCtrl.js (hunk 2 above)
Comment (vkartik97) [reply frontend/src/js/controllers/challengeCtrl.js]: Initialised with `True`, so required and to switch from 
- When Submissions are remaining **and** `Upload File` is selected  

**to** 
- When Submission Limit is exhausted

**Comment by RishabhJain2018** [comment]
@vkartik97 Can you please resolve the conflicts? Let's get this merged 🎉

**Comment by vkartik97** [comment]
@RishabhJain2018 Resolved conflicts!

File: frontend/src/js/controllers/challengeCtrl.js [hunk 3]
```diff
@@ -42,6 +42,7 @@
         vm.sortColumn = 'rank';
         vm.reverseSort = false;
         vm.columnIndexSort = 0;
+        vm.submissionsExhausted = true;
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeCtrl.js]: This variable name is confusing. Can you please change it to `disableSubmit`?

Comment (vkartik97) [reply frontend/src/js/controllers/challengeCtrl.js]: Done

**Pull request merged**